src/cdk-experimental/scrolling/virtual-for-of.ts

```diff
@@ -0,0 +1,330 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ArrayDataSource, CollectionViewer, DataSource, ListRange} from '@angular/cdk/collections';
+import {
+  Directive,
+  DoCheck,
+  EmbeddedViewRef,
+  Input,
+  IterableChangeRecord,
+  IterableChanges,
+  IterableDiffer,
+  IterableDiffers,
+  NgIterable,
+  OnDestroy,
+  SkipSelf,
+  TemplateRef,
+  TrackByFunction,
+  ViewContainerRef,
+} from '@angular/core';
+import {Observable, Subject} from 'rxjs';
+import {pairwise, shareReplay, startWith, switchMap, takeUntil} from 'rxjs/operators';
+import {CdkVirtualScrollViewport} from './virtual-scroll-viewport';
+
+
+/** The context for an item rendered by `CdkVirtualForOf` */
+export type CdkVirtualForOfContext<T> = {
+  $implicit: T;
+  cdkVirtualForOf: DataSource<T> | Observable<T[]> | NgIterable<T>;
+  index: number;
```

Comment:
Question: Is this the ACTUAL index or the index in view?

Reply:
This is the data index, added a comment to reflect that. We could add the view index too if we think its useful to people.